ammico/prompt_builder.py

```diff
@@ -0,0 +1,287 @@
+
+            The following are answers to above questions obtained for each segment of the video.
+            These answers are associated with the timestamp of each segment's beginning:"""
+        else:
+            header = """## FRAME-Level Answer Context (Reference Only)
+
+            For each question, the following are frame-level answers provided as reference.
+            If these answers are "Cannot be determined", do not accept that as final—instead, 
+            use visual and audio information to answer the question:"""
+
+        bullets_text = "\n".join(vqa_bullets)
+        return f"{header}\n\n{bullets_text}"
+
+    @classmethod
+    def build_frame_prompt(
+        cls, include_vqa: bool = False, questions: Optional[List[str]] = None
+    ) -> str:
+        """Build prompt for frame-level analysis."""
+        modules = [cls.ROLE_MODULE, cls.visual_frames_module()]
+
+        if include_vqa and not questions:
+            raise ValueError("Questions must be provided when VQA should be included.")
+
+        if include_vqa:
+            modules.append(cls.summary_vqa_task(ProcessingLevel.FRAME))
+            modules.append(cls.questions_module(questions))
+        else:
+            modules.append(cls.summary_task())
+
+        modules.append(cls.CONSTRAINTS_MODULE)
+        return "\n\n".join(modules)
+
+    @classmethod
+    def build_clip_prompt(
+        cls,
+        frame_bullets: List[str],
+        include_audio: bool = False,
+        audio_transcription: Optional[List[Dict]] = None,
+        include_vqa: bool = False,
+        questions: Optional[List[str]] = None,
+        vqa_bullets: Optional[List[str]] = None,
+    ) -> str:
+        """Build prompt for clip-level analysis."""
+        modules = [cls.ROLE_MODULE, cls.visual_captions_module(frame_bullets)]
+
+        if include_audio and audio_transcription:
+            modules.append(cls.audio_module(audio_transcription))
+
+        if include_vqa and not questions:
+            raise ValueError("Questions must be provided when VQA should be included.")
+
+        if include_vqa:
+            modules.append(
+                cls.summary_vqa_task(ProcessingLevel.CLIP, has_audio=include_audio)
+            )
+            modules.append(cls.questions_module(questions))
+            if vqa_bullets:
+                modules.append(cls.vqa_context_module(vqa_bullets, is_final=False))
+        else:
+            modules.append(cls.summary_task(has_audio=include_audio))
+
+        modules.append(cls.CONSTRAINTS_MODULE)
+        return "\n\n".join(modules)
+
+    @classmethod
+    def build_video_prompt(
+        cls,
+        summary_only: bool = False,
+        include_vqa: bool = False,
+        clip_summaries: Optional[List[str]] = None,
+        questions: Optional[List[str]] = None,
+        vqa_bullets: Optional[List[str]] = None,
+    ) -> str:
+        """Build prompt for video-level analysis."""
+        modules = [cls.ROLE_MODULE]
+
+        if summary_only:
```

Comment:
I am not sure why the default for `summary_only` is `False`. If both summary and question are not provided, nothing is done here. I suggest to adapt the behaviour of this method to the behaviour of the other methods, that is, `summary` is the default, if no questions provided raise error, and add an explanation why for the VQA bullets we can finally carry out the `vqa_only_task`.